specs/02-data-availability-sampling.md

```diff
@@ -0,0 +1,242 @@
+# Data Availability Sampling
+
+Child shards are propagating to the upper layers of the system commitments of their blocks through
+`BlockInfo`, and segment commitments to the beacon chain through `SegmentInfo` transactions.
+However, in order to be able to verify these blocks and segments, and to ensure that segments are
+always available for plotting, they all must be available for retrieval.
+
+This is done through a data availability sampling protocol that allows any node in the system to
+verify that blocks and segments are available for retrieval. The protocol is based on the idea of
+sampling a small number of chunks from encoded segments and blocks and verifying that they are
+available.
+
+### Availability sampling.
+
+This protocol is used to sample both: blocks and segments in shards.
+
+#### 1. Sampling process
+
+The sampling process is the same for both, blocks and segments. Any node in the system, but in
+particular light nodes, periodically sample random blocks and segments from different shards. We
+expect nodes of any kind to prioritise sampling the shards that they are interested in, but they are
+all free to choose where they sample.
+
+The sampling process is as follows:
+
+1.  **Select a Chunk**:
+    - Choose a random share index `i` from `0` to `total_shares - 1`.
+    - Choose a random chunk index `j` from `0` to `num_chunks_per_share - 1`.
+2.  **Request Data**:
+    - Request chunk `j` from share `i` and its witness `π_{i,j}` from a node (e.g., farmer).
+    - The witness should include at least:
+      - Intra-share Merkle path from chunk `j` to `Cbs_i` (`Cs_i` for segments).
+      - Inter-share Merkle path from `Cbs_i` to `C_block` (`Cs_i` and `segment_commitment` for
+        segments).
+    - Segment sample verification can also optionally include a witness to verify that the segment
+      is included in the global history in the beacon chain through the `super_segment_commitment`
+      for that shard segment and its corresponding witness.
+    - Requests are broadcast to the network, they are not directed to a specific node.
+3.  **Repeat**:
+    - Perform multiple sampling rounds to build confidence.
+
+#### 2. Verification
+
+For each sampled chunk:
+
+- Verify the witness `π_{i,j}` by:
+  - Hashing the chunk data with the intra-share Merkle path to compute `Cbs_i` (`Cs_i` for
+    segments).
+  - Hashing `Cbs_i` with the inter-share Merkle path to compute `C_block` (`Cs_i` and
+    `segment_commitment` for segments).
+- Compare the computed `C_block` with the trusted `C_block` from the `BlockInfo` on the parent
+  chain. For segments, compare with the `segment_commitment` from the `SegmentInfo` transaction in
+  the beacon chain.
+- **Outcome**:
+  - **Success**: Chunk is available and authentic.
+  - **Failure**: Chunk is missing or the proof is invalid, indicating potential unavailability.
+- All outcomes are signed and gossiped to the network so nodes can leverage information from others
```

Comment:
I'm not sure what purpose signing is supposed to serve, a signature from unknown identity doesn't really help with anything.